cmd/csvlint/main.go

```diff
@@ -22,16 +24,17 @@ func main() {
 		printHelpAndExit(0)
 	}
 
-	var comma rune
-	switch *delimiter {
-	case "comma":
-		comma = ','
-	case "tab":
-		comma = '\t'
-	default:
-		fmt.Printf("unrecognized delimiter '%s'\n\n", *delimiter)
+	if flag.NFlag() > 0 {
+		fmt.Fprintln(os.Stderr, "Warning: not using defaults, may not validate CSV to RFC 4180")
+	}
+
+	converted_delimiter, err := strconv.Unquote(`'` + *delimiter + `'`)
```

Comment:
fixing the linter: 4b93977b64d836d3f9adc9b66c3be9d521fe9d88
fixing the underscore: bf74cef0a18e34040d48ecfc8c7dcad0f5da716e
I've noticed that the `gofmt` command has the same problem - it does not recursively lint to `cmd/csvlint/main.go` as advertised. Tried the same wildcard path and it didn't work... we can take a look tomorrow.